### ISSUE TITLE
Inconsistent closing of settings panels

### ISSUE THREAD

**Issue by maddisondesigns** [WordPress/gutenberg#18418]
**Describe the bug**
The panels in the Settings Sidebar are inconsistent on whether they stay open or stay closed when moving between blocks.

Here's a video of six different Heading blocks. When I click between each of them the _Color Settings_ panel constantly closes, but the _Advanced_ panel constantly stays open. It's inconsistent, and extremely annoying when they constantly close.

If I open one panel (the _Advanced_ panel), and it stays open as I move to other blocks, then the same thing should happen when I open another panel (the _Color Settings_ panel), but it doesn't.

Vid: https://cl.ly/b500cff8e9e0

This issues happens on numerous blocks such as the Heading Block and the Paragraph Block (and most likely other blocks, but I haven't checked every single block)

**To reproduce**
1. Add multiple Heading blocks
2. Open the  _Color Settings_ and _Advanced_ panels in the Settings Sidebar
3. Select each Block alternately and watch the  _Color Settings_ panel constantly close

**Expected behavior**
Settings Sidebar panels are inconsistent in how they display.

**Desktop (please complete the following information):**
- OS: macOS Mojave 10.14.6
- Browser: Firefox 70.0.1
- Version: WP 5.3-RC4-46680

**Comment by karmatosed** [comment]
I can replicate this, thanks for discovering this as always great to soothe inconsistencies. I think this needs a decision on what stays open. My gut says if it's the same block they all should, color included.

**Comment by karmatosed** [comment]
Noting this as good first issue and that making it consistent would be issue.

**Comment by juanfra** [comment]
I'd love to help on this one. I started to make some tests and the only thing I found (If I understood correctly the way it should be implemented) is that once we have the two buttons one beside the other, when "Publish" is clicked, the "Cancel" button is just beside, but the action won't cancel the publishing, but close the panel instead. I found this the only thing that could be misleading:

<img src="https://cldup.com/iD8AUtDNfL.gif" >

**Comment by skorasaurus** [comment]
heavily related issues: https://github.com/WordPress/gutenberg/issues/27560

**Comment by Saoming** [comment]
Can I work on this issue?

**Comment by skorasaurus** [comment]
> Can I work on this issue?

Yes; although if you submit a pull request it may take some time to someone review it.

**Comment by Saoming** [comment]
All good, Thank you @skorasaurus! I am not sure how to assign myself here

**Comment by richtabor** [comment]
I actually don't think this is as big of an issue, as many of the default panels are using toolbelt controls (the vertical ellipsis) and remain open by default.  

https://user-images.githubusercontent.com/1813435/208713036-74a71ce7-4404-47f7-b5f5-1b37fff52040.mp4

**Comment by maddisondesigns** [comment]
It's probably worth pointing out that this issue was raised 3+ years ago so a lot has changed since then. Although a lot of the Settings Sidebar panels now remain open, they are still wildly inconsistent.  Why do some panels allow you to open/close them while others don't?

From what I can tell, it's due to the poor design of some of the controls. Controls like Typography, Colour & Dimensions, for example, require you to "turn on" various options before you can even use them, which is super annoying. The Options icon for these controls is sitting in the same position as the open/close toggle on other panels. You shouldn't have to enable options prior to being able to use them, they should all be available, all the time. No only that, but the fact that enabling them on one block, doesn't automatically enable them on all the other same blocks, is annoying as well. Editing content in Gutenberg is so time consuming and slow because you hide so many controls and options by default, in hidden panels and popups, forcing users to perform multiple unnecessary clicks.

Vid: https://share.getcloudapp.com/YEuDXY8L

**Comment by Saoming** [comment]
I agree with you @maddisondesigns , Maybe it should reduce the toggle

**Comment by rdhutchins** [comment]
So many years have passed and we STILL have to click "+" to even see the margin/padding options. Why is this? Is this a purposeful design decision, and what is the rationale? Isn't there at least an editor hook we could use in code to default these to open? I have searched and searched.
